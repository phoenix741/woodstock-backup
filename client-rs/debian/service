[Unit]
Description=Woodstock Client Daemon

[Service]
ExecStart=/usr/bin/ws_client_daemon

[Install]
WantedBy=network.target