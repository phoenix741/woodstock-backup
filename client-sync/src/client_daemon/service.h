#pragma once

#include <QtService>

#include "woodstockclient.h"

class ClientService : public QtService::Service
{
    Q_OBJECT
public:
    explicit ClientService(int &argc, char **argv);

protected:
    virtual ClientService::CommandResult onStart() override;
    virtual ClientService::CommandResult onStop(int &exitCode) override;

private:
    std::unique_ptr<WoodstockClient> m_client;
};
