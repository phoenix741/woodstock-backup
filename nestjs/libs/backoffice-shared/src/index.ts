export * from './backoffice-shared.module';
export * from './config';
export * from './logger';
export * from './models';
export * from './services';
