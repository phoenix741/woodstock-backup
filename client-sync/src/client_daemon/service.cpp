#include "service.h"

ClientService::ClientService(int &argc, char **argv) : Service{argc, argv}, m_client(new WoodstockClient())
{
}

ClientService::CommandResult ClientService::onStart()
{
    qDebug() << "Service was started";

    m_client->listen();

    qDebug() << "Service is listening";
    return ClientService::CommandResult::Completed; //service is now assumed started
}

ClientService::CommandResult ClientService::onStop(int &exitCode)
{
    qDebug() << "Stop received";
    
    exitCode = 0;

    m_client->stop();

    qDebug() << "Stopped";
    return ClientService::CommandResult::Completed;
}
