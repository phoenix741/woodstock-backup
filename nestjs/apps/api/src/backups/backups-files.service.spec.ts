import { Test, TestingModule } from '@nestjs/testing';
import { BackupsService } from '@woodstock/backoffice-shared';
import * as mock from 'mock-fs';
import { BackupsFilesService } from './backups-files.service';

describe('Backups File Service', () => {
  let service: BackupsFilesService;

  const mockBackupsService = {
    getDestinationDirectory: (name: string, number: number) => `destinationDirectory/${name}/${number}`,
  };

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [{ provide: BackupsService, useValue: mockBackupsService }, BackupsFilesService],
    }).compile();

    service = module.get<BackupsFilesService>(BackupsFilesService);
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });

  it('should list all share of a directory', async () => {
    // GIVEN
    mock({
      'destinationDirectory/name/10': {
        file1: mock.file({
          content: 'file content here',
          atime: new Date(1),
          ctime: new Date(1),
          mtime: new Date(1),
          birthtime: new Date(1),
          uid: 0,
          gid: 0,
        }),
        dir1: mock.directory({
          atime: new Date(1),
          ctime: new Date(1),
          mtime: new Date(1),
          birthtime: new Date(1),
          uid: 0,
          gid: 0,
          items: {
            /** empty directory */
          },
        }),
      },
    });

    // WHEN
    const listShare = await service.listShare('name', 10);

    // THEN
    expect(listShare[0]).toMatchSnapshot(
      {
        ino: expect.any(Number),
      },
      'listShare[0]',
    );

    expect(listShare[1]).toMatchSnapshot(
      {
        ino: expect.any(Number),
      },
      'listShare[1]',
    );
  });

  it('should list all files of a directory', async () => {
    // GIVEN
    mock({
      'destinationDirectory/name/10/%2Fshare/toto': {
        file1: mock.file({
          content: 'file content here',
          atime: new Date(1),
          ctime: new Date(1),
          mtime: new Date(1),
          birthtime: new Date(1),
          uid: 0,
          gid: 0,
        }),
        dir1: mock.directory({
          atime: new Date(1),
          ctime: new Date(1),
          mtime: new Date(1),
          birthtime: new Date(1),
          uid: 0,
          gid: 0,
          items: {
            /** empty directory */
          },
        }),
      },
    });

    // WHEN
    const listFiles = await service.list('name', 10, '/share', '/toto');

    // THEN
    expect(listFiles[0]).toMatchSnapshot(
      {
        ino: expect.any(Number),
      },
      'listFiles[0]',
    );

    expect(listFiles[1]).toMatchSnapshot(
      {
        ino: expect.any(Number),
      },
      'listFiles[1]',
    );
  });

  it('should throw exception when list of file is not absolute ', async () => {
    expect.assertions(1);
    await expect(service.list('name', 10, '/share', 'toto')).rejects.toThrow(
      'Only absolute path can be used to serach for directory',
    );
  });

  it('should throw exception when list of share is not absolute ', async () => {
    expect.assertions(1);
    await expect(service.list('name', 10, 'share', '/toto')).rejects.toThrow(
      'Only absolute path can be used to serach for directory',
    );
  });

  it('should get the file stat of a filename', async () => {
    // GIVEN
    mock({
      'destinationDirectory/name/10/%2Fshare/toto': {
        file1: mock.file({
          content: 'file content here',
          atime: new Date(1),
          ctime: new Date(1),
          mtime: new Date(1),
          birthtime: new Date(1),
          uid: 0,
          gid: 0,
        }),
      },
    });

    // WHEN
    const infos = await service.getFileName('name', 10, '/share', '/toto/file1');

    // THEN
    expect(infos).toMatchSnapshot(
      {
        stats: {
          ino: expect.any(Number),
        },
      },
      'service.getFileName',
    );
  });

  it('should throw exception when list of file is not absolute ', async () => {
    expect.assertions(1);
    await expect(service.list('name', 10, '/share', 'toto')).rejects.toThrow(
      'Only absolute path can be used to serach for directory',
    );
  });

  it('should throw exception when list of share is not absolute ', async () => {
    expect.assertions(1);
    await expect(service.list('name', 10, 'share', '/toto')).rejects.toThrow(
      'Only absolute path can be used to serach for directory',
    );
  });

  afterEach(() => {
    mock.restore();
  });
});
