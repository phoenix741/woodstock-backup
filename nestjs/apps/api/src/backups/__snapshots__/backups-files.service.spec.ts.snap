// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backups File Service should get the file stat of a filename: service.getFileName 1`] = `
Object {
  "filename": "destinationDirectory/name/10/%2Fshare/toto/file1",
  "stats": Object {
    "atime": 1970-01-01T00:00:00.001Z,
    "atimeMs": 1,
    "birthtime": 1970-01-01T00:00:00.001Z,
    "birthtimeMs": 1,
    "blksize": 4096,
    "blocks": 1,
    "ctime": 1970-01-01T00:00:00.001Z,
    "ctimeMs": 1,
    "dev": 8675309,
    "gid": 0,
    "ino": Any<Number>,
    "mode": 33206,
    "mtime": 1970-01-01T00:00:00.001Z,
    "mtimeMs": 1,
    "nlink": 1,
    "rdev": 0,
    "size": 17,
    "uid": 0,
  },
}
`;

exports[`Backups File Service should list all files of a directory: listFiles[0] 1`] = `
Object {
  "atime": 1970-01-01T00:00:00.001Z,
  "atimeMs": 1,
  "birthtime": 1970-01-01T00:00:00.001Z,
  "birthtimeMs": 1,
  "blksize": 4096,
  "blocks": 1,
  "ctime": 1970-01-01T00:00:00.001Z,
  "ctimeMs": 1,
  "dev": 8675309,
  "gid": 0,
  "ino": Any<Number>,
  "mode": 16895,
  "mtime": 1970-01-01T00:00:00.001Z,
  "mtimeMs": 1,
  "name": Object {
    "data": Array [
      100,
      105,
      114,
      49,
    ],
    "type": "Buffer",
  },
  "nlink": 2,
  "rdev": 0,
  "size": 1,
  "type": "DIRECTORY",
  "uid": 0,
}
`;

exports[`Backups File Service should list all files of a directory: listFiles[1] 1`] = `
Object {
  "atime": 1970-01-01T00:00:00.001Z,
  "atimeMs": 1,
  "birthtime": 1970-01-01T00:00:00.001Z,
  "birthtimeMs": 1,
  "blksize": 4096,
  "blocks": 1,
  "ctime": 1970-01-01T00:00:00.001Z,
  "ctimeMs": 1,
  "dev": 8675309,
  "gid": 0,
  "ino": Any<Number>,
  "mode": 33206,
  "mtime": 1970-01-01T00:00:00.001Z,
  "mtimeMs": 1,
  "name": Object {
    "data": Array [
      102,
      105,
      108,
      101,
      49,
    ],
    "type": "Buffer",
  },
  "nlink": 1,
  "rdev": 0,
  "size": 17,
  "type": "REGULAR_FILE",
  "uid": 0,
}
`;

exports[`Backups File Service should list all share of a directory: listShare[0] 1`] = `
Object {
  "atime": 1970-01-01T00:00:00.001Z,
  "atimeMs": 1,
  "birthtime": 1970-01-01T00:00:00.001Z,
  "birthtimeMs": 1,
  "blksize": 4096,
  "blocks": 1,
  "ctime": 1970-01-01T00:00:00.001Z,
  "ctimeMs": 1,
  "dev": 8675309,
  "gid": 0,
  "ino": Any<Number>,
  "mode": 16895,
  "mtime": 1970-01-01T00:00:00.001Z,
  "mtimeMs": 1,
  "name": Object {
    "data": Array [
      100,
      105,
      114,
      49,
    ],
    "type": "Buffer",
  },
  "nlink": 2,
  "rdev": 0,
  "size": 1,
  "type": "DIRECTORY",
  "uid": 0,
}
`;

exports[`Backups File Service should list all share of a directory: listShare[1] 1`] = `
Object {
  "atime": 1970-01-01T00:00:00.001Z,
  "atimeMs": 1,
  "birthtime": 1970-01-01T00:00:00.001Z,
  "birthtimeMs": 1,
  "blksize": 4096,
  "blocks": 1,
  "ctime": 1970-01-01T00:00:00.001Z,
  "ctimeMs": 1,
  "dev": 8675309,
  "gid": 0,
  "ino": Any<Number>,
  "mode": 33206,
  "mtime": 1970-01-01T00:00:00.001Z,
  "mtimeMs": 1,
  "name": Object {
    "data": Array [
      102,
      105,
      108,
      101,
      49,
    ],
    "type": "Buffer",
  },
  "nlink": 1,
  "rdev": 0,
  "size": 17,
  "type": "REGULAR_FILE",
  "uid": 0,
}
`;
